models/articles.models.js

```diff
@@ -15,13 +20,15 @@ exports.fetchArticles = (sort_by = "created_at", order_by = "DESC") => {
       message: "can't order by this query",
     });
   }
-  return db
-    .query(
-      `SELECT articles.article_id, articles.title, articles.topic, articles.author, articles.created_at, articles.votes, articles.article_img_url, COUNT(comments.comment_id) AS comment_count FROM comments LEFT JOIN articles ON comments.article_id = articles.article_id GROUP BY articles.article_id ORDER BY ${sort_by} ${order_by}`
-    )
-    .then(({ rows }) => {
+  if (topic !== "topic") {
+    queryStr = `SELECT articles.article_id, articles.title, articles.topic, articles.author, articles.created_at, articles.votes, articles.article_img_url, COUNT(comments.comment_id) AS comment_count FROM comments LEFT JOIN articles ON comments.article_id = articles.article_id WHERE topic=$1 GROUP BY articles.article_id ORDER BY ${sort_by} ${order_by}`;
+    return db.query(queryStr, [topic]).then(({ rows }) => {
```

Comment:
You can avoid the need for this entire if statement by removing the default value for topic